src/components/GoalHeader/GoalHeader.tsx

```diff
@@ -19,12 +20,12 @@ interface GoalHeaderProps
     goal?: Partial<GoalByIdReturnType>;
     actions?: ReactNode;
 
-    onGoalStateChange?: ComponentProps<typeof StateDropdown>['onChange'];
+    onGoalStateChange?: (value: GoalStateChangeSchema['state']) => void;
```

Comment:
But why?

Reply:
return value of `onChange` depends on the mode prop. 

substituting a specific type such as ComponentProps<typeof StateDropdown>['onChange'] ensures the correct usage of the function and simplifies understanding of its operation.